packages/frontend-core/src/components/grid/stores/viewV2.js

```diff
@@ -0,0 +1,216 @@
+import { get } from "svelte/store"
+
+const SuppressErrors = true
+
+export const createActions = context => {
+  const { definition, API, datasource, columns, stickyColumn } = context
+
+  const refreshDefinition = async () => {
+    const $datasource = get(datasource)
+    if (!$datasource) {
+      definition.set(null)
+      return
+    }
+    const table = await API.fetchTableDefinition($datasource.tableId)
+    const view = Object.values(table?.views || {}).find(
+      view => view.id === $datasource.id
+    )
+    definition.set(view)
+  }
+
+  const saveDefinition = async newDefinition => {
+    await API.viewV2.update(newDefinition)
+  }
+
+  const saveRow = async row => {
+    const $datasource = get(datasource)
+    row.tableId = $datasource?.tableId
+    row._viewId = $datasource?.id
+    return {
+      ...(await API.saveRow(row, SuppressErrors)),
+      _viewId: row._viewId,
+    }
+  }
+
+  const deleteRows = async rows => {
+    // Ensure we delete _viewId from rows as otherwise this throws a 500
```

Comment:
Not sure if this should be the case anymore

Reply:
I'll check and remove this if no longer needed.

Reply:
No longer needed in the store because the API client has this deletion code, so I can remove it from here indeed - nice catch.

The logic is still needed is general though, as if we do not remove `_viewId` then the rows API will throw a 400 with `Bad special document member _viewId`.